Lib/test/test_builtin.py

```diff
@@ -2138,6 +2138,24 @@ def test_bool_notimplemented(self):
         with self.assertRaisesRegex(TypeError, msg):
             not NotImplemented
 
+    def test_singleton_attribute_access(self):
+        for singleton in (NotImplemented, Ellipsis):
+            with self.subTest(singleton):
+                self.assertIs(type(singleton), singleton.__class__)
+                self.assertIs(type(singleton).__class__, type)
+
+                # Missing instance attributes:
+                with self.assertRaises(AttributeError):
+                    singleton.prop
+                with self.assertRaises(AttributeError):
+                    singleton.prop = 1
```

Comment:
I would put the setter tests first (here and in the next blocks). This guards better against potential bugs where the setter raises an exception but sets the attribute anyway.